### PR TITLE
Lame duck mode support

### PR DESCRIPTION
**Pull request by Jarema**
Realized that we should probably do more when new info arrives, hence used `process_info` instead of more specific name for lame duck tape mode.

### CODE REVIEW

File: src/connect.rs [hunk 1]
```diff
@@ -68,6 +74,15 @@ pub struct ConnectInfo {
     pub no_responders: bool,
 }
 
+/// Protocol version used by the client.
+#[derive(Clone, Copy, Debug)]
+pub enum Protocol {
+    /// Original protocol.
+    Original = 0,
+    /// Prptocol with dynamic reconfiguration of cluster and lame duck mode functionality.
```
Comment (caspervonb) [inline src/connect.rs:82]:
```suggestion
    /// Protocol with dynamic reconfiguration of cluster and lame duck mode functionality.
```

**Review by caspervonb — COMMENTED** [review]
A test would be great if we can reliably test this 🙂

**Comment by Jarema** [comment]
Disabled lame duck mode test for Windows as it requires Windows service (lame duck mode test runs `nats-server` command with signal).

**Review by caspervonb — APPROVED** [review]
lgtm

**Pull request merged**